02_bubble/harirakr/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	n := len(s)
+	cs := append([]int{}, s...)
+	for i := n; i > 0; i-- {
+		for j := 0; j < i-1; j++ {
+			if cs[j] > cs[j+1] {
```

Comment:
Please see if you can optimize the bubble sort to avoid iteration if the array is already sorted

Reply:
Don.